features/steps/thens/attributes.py

```diff
@@ -143,3 +143,169 @@ def step_impl(context, inst):
+
+    operators = {
+        '+' : operator.add,
+        '-' : operator.sub,
+        '*' : operator.mul,
+        '/' : operator.truediv,
+        '%' : operator.mod,
+        '**' : operator.pow,
+        'equal to' : operator.eq,
+        'not equal to' : operator.ne,
+        'greater than' : operator.gt,
+        'less than' : operator.gt,
+        'greater than or equal to' : operator.ge,
+        'less than or equal to' : operator.le,
+    }
+
+    # Get compared attribute value
+
+    attr_compared_value = getattr(inst, attribute_compared, 'Compared attribute not found')
+    if isinstance(attr_compared_value, ifcopenshell.entity_instance):
+        raise Exception('Compared attribute value is an IFC entity')
+
+    # Replace attribute names with attribute values in the expression
+    for string in expression.split():
+        # Checks if the string is not a operator neither parenthesis
+        if string not in [*operators, '(', ')']:
+            if hasattr(inst, string):
+                if not isinstance(getattr(inst, string), ifcopenshell.entity_instance):
+                    expression = expression.replace(string, str(getattr(inst, string)))
+                else:
+                    raise Exception('Expression attribute value is an IFC entity')
+            else:
+                raise Exception('Expression attribute not found')
+
+    # Evaluate the string expression using eval
+    try:
+        expression_value = eval(expression)
+    except Exception as e:
+        raise ValueError(f"Error evaluating expression: {e}")
+
+    # Compare the attribute with the expression value, considering the precision
+    entity_contexts = geometry.recurrently_get_entity_attr(context, inst, 'IfcRepresentation', 'ContextOfItems')
+    precision = geometry.get_precision_from_contexts(entity_contexts)
+
+    try:
+        result = geometry.compare_with_precision(attr_compared_value, expression_value, precision, comparison_op)
+        if result:
+            yield ValidationOutcome(
+                inst=inst,
+                expected=f"A value {comparison_op} {expression_value} with precision {precision}",
+                observed={attr_compared_value},
+                severity=OutcomeSeverity.PASSED,
+            )
+        else:
+            yield ValidationOutcome(
+                inst=inst,
+                expected=f"A value {comparison_op} {expression_value}",
+                observed={attr_compared_value},
+                severity=OutcomeSeverity.ERROR,
+            )
+    except ValueError as e:
+        raise ValueError(f"Error during comparison: {e}")
+
+@gherkin_ifc.step('First instance {first_expression} value must be {comparison_op} the second instance {second_expression}')
+@gherkin_ifc.step('First instance {first_expression} value must be {comparison_op} the second instance {second_expression} at depth 1')
+def step_impl(context, inst, first_expression:str, comparison_op:str, second_expression:str):
```

Comment:
I don't see a usage of this impl function

Reply:
Correct, and thanks for the catch.  We originally had a number of additional scenarios and this one handles the check of `StartDistAlong` that I mentioned in Slack.  See also IVS-150 in JIRA.  I would like to leave it in for now so that it's there fore future rules.